File: vertex/vector.go
// Go
package main

import (
	"fmt"
	"math"
)

type Vector struct {
	x, y float64
}

func (v Vector) Magnitude() float64 {
	return math.Sqrt(v.x*v.x + v.y*v.y)
}

func main() {
	
	v1 := Vector{3, 4}
	v2 := Vector{1, 2}
	
	fmt.Println(v1.Magnitude()) // -> 5
	fmt.Println(v2.x)        // -> 1
}

File: javascript/data-structures/vector.js
function vector(x, y) {
    return Object.freeze({
        get point() {
            return [x, y];
        },
        get magnitude() {
            return Math.sqrt(x * x + y * y);
        },
        scale(factor) {
            x *= factor;
            y *= factor;
        },
    });
}
